templates/test/live.html.ep

```diff
@@ -1,24 +1,92 @@
-<div class='container'>
-  <div class='row'>
-    <div class='col-sm-3'>
-      Running Module:
+% content_for 'ready_function' => begin
+    % if (my $current_user = current_user) {
+        window.ownUserId = <%= $current_user->id %>;
+        window.isAdmin = <%= ($current_user->is_admin) ? ('true') : ('false') %>;
+    % }
+    setupDeveloperPanel();
+% end
+
+<div class="card card-outline-secondary filter-panel-bottom" id="developer-panel" data-developer-url="<%= $ws_developer_url %>" data-status-only-url="<%= $ws_status_only_url %>">
+    <div class="card-header">
+        <strong>Developer mode</strong>
+        <span>
+            <span class="developer-mode-element" data-hidden-on="isConnected">
+                retrieving status <i class="fas fa-spinner fa-spin fa-lg"></i>
+            </span>
+            <span class="developer-mode-element" id="developer-status-info" data-visible-on="isConnected">
+                unknown status
+            </span>
+        </span>
+        <div class="card-header-appendix" id="developer-session-info">
+            % if (my $developer_session = $job->developer_session) {
+                % my $tab_count = $developer_session->ws_connection_count;
+                opened by <%= $developer_session->user->name %>
+                (started
+                    <abbr class="timeago" title="<%= $developer_session->t_created->datetime() %>Z">
+                        %= format_time($developer_session->t_created)
+                    </abbr>,
+                developer has <%= $tab_count %> browser <%= $tab_count == 1 ? 'tab' : 'tabs' %> open)
+            % }
+            % else {
+                no developer session opended
+                % if (is_admin) {
+                    - click to open
+                % }
+            % }
+        </div>
     </div>
-    <div id="running_module" class='col-sm-9'>Unknown</div>
-  </div>
-  % if (my $developer_session = $job->developer_session) {
-    <div class='row'>
-        <div class='col-sm-3'>
-        Developer session:
+    <div class="card-body">
+        <form action="#" method="get" id="developer-form" class="developer-mode-element" data-hidden-on="lockedByOtherDeveloper">
+            <p class="developer-mode-element" data-hidden-on="ownSession">
+                Select what you want to do and confirm using the button below. This will start a developer
```

Comment:
@cyntss Can you give your blessing here on the UI/UX side of things? :). There are some screenshots attached, but you can always do your own run. /me wonders if mail would work better :P

Reply:
it makes sense now that I see it in openQA.